src/asset-publisher.ts

```diff
@@ -38,15 +39,23 @@ export class AssetPublisher {
         }
         return config;
     }
-    constructor(rootPath: string, s3: S3, cfg: Config) {
+    constructor(rootPath: string, s3: S3, cfg: Config, staticSite = false) {
         this.cfg = cfg;
         this.rootPath = rootPath;
         this.s3 = s3;
+        this.staticSite = staticSite
     }
 
     public async deployStaticAssets(subdir = 'out') {
         const waitList = [];
-        const manifest = new ManifestBuilder(this.rootPath).loadManifest();
+
+        let manifest;
+        if(this.staticSite) {
```

Comment:
`staticSite` being a boolean indicates that we could have a base `AssetPublisher` and specialize it overloading the `deployStaticAssets()` method.